### ISSUE TITLE
16.09 crashed after Windows Service start

### ISSUE THREAD

**Issue by ashum81** [processone/ejabberd#1374]
> What version of ejabberd are you using?

ejabberd Community Server 16.09 windows 64bit installer

> What operating system (version) are you using?

Windows 2012 R2 standard, Microsoft Windows [Version 6.3.9600]

> How did you install ejabberd (source, package, distribution)?

Windows 64bits Installer

> What did not work as expected? Are there error messages in the log? What
> was the unexpected behavior? What was the expected result?

I create win service with a script:
```
erlsrv add ejabberd -sname ejabberd@localhost -w "C:\Program Files\ejabberd-16.09\bin"
erlsrv set ejabberd -ar "-kernel inetrc '\"%APPDATA%/ejabberd/conf/inetrc\"' -mnesia dir '\"%APPDATA%/ejabberd/database/mnesia\"' -sasl sasl_error_logger {file,'\"C:/Program Files/ejabberd-16.09/logs/sasl.log\"'} -setcookie ejabberd -s ejabberd -smp enable +P 250000"
erlsrv set ejabberd -e "EJABBERD_BIN_PATH=C:\Program Files\ejabberd-16.09\lib\ejabberd-16.09\priv\bin"
erlsrv set ejabberd -e "EJABBERD_CONFIG_PATH=%APPDATA%\ejabberd\conf\ejabberd.yml"
erlsrv set ejabberd -e "EJABBERD_DOC_PATH=C:\Program Files\ejabberd-16.09\doc"
erlsrv set ejabberd -e "EJABBERD_LOG_PATH=%APPDATA%\ejabberd\logs\ejabberd.log"
erlsrv set ejabberd -e "EJABBERD_MSGS_PATH=C:\Program Files\ejabberd-16.09\lib\ejabberd-16.09\priv\msgs"
erlsrv set ejabberd -e "ERL_CRASH_DUMP=%APPDATA%\ejabberd\logs\erl_crash.dump"
erlsrv set ejabberd -e "ERL_INETRC=%APPDATA%\ejabberd\conf\inetrc"
erlsrv set ejabberd -e "ERL_LIBS=C:\Program Files\ejabberd-16.09\lib"
erlsrv set ejabberd -e "ERL_MAX_ETS_TABLES=1400"
erlsrv set ejabberd -e "ERL_MAX_PORTS=32000"
```
 The service started but ejabberd crashed with error:
error.log
```
2016-11-09 14:15:50.025 [error] <0.63.0>@lager_handler_watcher:123 Lager fatally failed to install handler lager_console_backend into lager_event, NOT retrying: old_shell
2016-11-09 14:15:53.353 [error] <0.40.0> CRASH REPORT Process <0.40.0> with 0 neighbours exited with reason: {aborted,{no_exists,passwd,attributes}} in application_master:init/4 line 134
```

ejabberd.log
```
2016-11-09 14:15:50.025 [warning] <0.58.0>@lager_console_backend:66 Lager's console backend is incompatible with the 'old' shell, not enabling it
2016-11-09 14:15:50.025 [error] <0.63.0>@lager_handler_watcher:123 Lager fatally failed to install handler lager_console_backend into lager_event, NOT retrying: old_shell
2016-11-09 14:15:50.056 [info] <0.7.0> Application lager started on node ejabberd@localhost
2016-11-09 14:15:50.087 [info] <0.7.0> Application crypto started on node ejabberd@localhost
2016-11-09 14:15:50.103 [info] <0.7.0> Application sasl started on node ejabberd@localhost
2016-11-09 14:15:50.118 [info] <0.7.0> Application asn1 started on node ejabberd@localhost
2016-11-09 14:15:50.118 [info] <0.7.0> Application public_key started on node ejabberd@localhost
2016-11-09 14:15:50.150 [info] <0.7.0> Application ssl started on node ejabberd@localhost
2016-11-09 14:15:50.181 [info] <0.7.0> Application fast_yaml started on node ejabberd@localhost
2016-11-09 14:15:50.212 [info] <0.7.0> Application fast_tls started on node ejabberd@localhost
2016-11-09 14:15:50.259 [info] <0.7.0> Application fast_xml started on node ejabberd@localhost
2016-11-09 14:15:50.259 [info] <0.7.0> Application stringprep started on node ejabberd@localhost
2016-11-09 14:15:50.306 [info] <0.7.0> Application cache_tab started on node ejabberd@localhost
2016-11-09 14:15:50.368 [info] <0.7.0> Application elixir started on node ejabberd@localhost
2016-11-09 14:15:50.556 [info] <0.7.0> Application mnesia started on node ejabberd@localhost
2016-11-09 14:15:51.806 [info] <0.7.0> Application inets started on node ejabberd@localhost
2016-11-09 14:15:53.322 [notice] <0.58.0>@lager_file_backend:131 Changed loglevel of C:\Users\as\AppData\Roaming\ejabberd\logs\ejabberd.log to info
2016-11-09 14:15:53.353 [info] <0.41.0>@cyrsasl_digest:start:60 FQDN used to check DIGEST-MD5 SASL authentication: <<"b3">>
2016-11-09 14:15:53.353 [info] <0.41.0>@ejabberd_app:add_windows_nameservers:165 Adding machine's DNS IPs to Erlang system:
[]
2016-11-09 14:15:53.353 [error] <0.40.0> CRASH REPORT Process <0.40.0> with 0 neighbours exited with reason: {aborted,{no_exists,passwd,attributes}} in application_master:init/4 line 134
2016-11-09 14:15:53.353 [info] <0.7.0> Application ejabberd exited with reason: {aborted,{no_exists,passwd,attributes}}
```

From desktop link it starts successfully.

**Comment by cromain** [comment]
there is a possibility ejabberd can not write to %APPDATA%/ejabberd when started as a service.
Can you copy %APPDATA%/ejabberd to %ProgramData%/ejabberd and change your script to use %programData% instead ?

**Comment by ashum81** [comment]
I have copied ejabberd folder to ProgramData. The same error:

error.log
```
2016-11-28 14:52:46.362 [error] <0.63.0>@lager_handler_watcher:123 Lager fatally failed to install handler lager_console_backend into lager_event, NOT retrying: old_shell
2016-11-28 14:52:50.830 [error] <0.40.0> CRASH REPORT Process <0.40.0> with 0 neighbours exited with reason: {aborted,{no_exists,passwd,attributes}} in application_master:init/4 line 134
```

ejabberd.log
```
2016-11-28 14:52:46.362 [warning] <0.58.0>@lager_console_backend:66 Lager's console backend is incompatible with the 'old' shell, not enabling it
2016-11-28 14:52:46.362 [error] <0.63.0>@lager_handler_watcher:123 Lager fatally failed to install handler lager_console_backend into lager_event, NOT retrying: old_shell
2016-11-28 14:52:46.362 [info] <0.7.0> Application lager started on node ejabberd@localhost
2016-11-28 14:52:46.393 [info] <0.7.0> Application crypto started on node ejabberd@localhost
2016-11-28 14:52:46.408 [info] <0.7.0> Application sasl started on node ejabberd@localhost
2016-11-28 14:52:46.440 [info] <0.7.0> Application asn1 started on node ejabberd@localhost
2016-11-28 14:52:46.440 [info] <0.7.0> Application public_key started on node ejabberd@localhost
2016-11-28 14:52:46.455 [info] <0.7.0> Application ssl started on node ejabberd@localhost
2016-11-28 14:52:46.487 [info] <0.7.0> Application fast_yaml started on node ejabberd@localhost
2016-11-28 14:52:46.533 [info] <0.7.0> Application fast_tls started on node ejabberd@localhost
2016-11-28 14:52:46.580 [info] <0.7.0> Application fast_xml started on node ejabberd@localhost
2016-11-28 14:52:46.580 [info] <0.7.0> Application stringprep started on node ejabberd@localhost
2016-11-28 14:52:46.627 [info] <0.7.0> Application cache_tab started on node ejabberd@localhost
2016-11-28 14:52:46.690 [info] <0.7.0> Application elixir started on node ejabberd@localhost
2016-11-28 14:52:46.877 [info] <0.7.0> Application mnesia started on node ejabberd@localhost
2016-11-28 14:52:48.330 [info] <0.7.0> Application inets started on node ejabberd@localhost
2016-11-28 14:52:50.783 [notice] <0.58.0>@lager_file_backend:131 Changed loglevel of C:\ProgramData\ejabberd\logs\ejabberd.log to info
2016-11-28 14:52:50.815 [info] <0.41.0>@cyrsasl_digest:start:60 FQDN used to check DIGEST-MD5 SASL authentication: <<"b3">>
2016-11-28 14:52:50.830 [info] <0.41.0>@ejabberd_app:add_windows_nameservers:165 Adding machine's DNS IPs to Erlang system:
[]
2016-11-28 14:52:50.830 [error] <0.40.0> CRASH REPORT Process <0.40.0> with 0 neighbours exited with reason: {aborted,{no_exists,passwd,attributes}} in application_master:init/4 line 134
2016-11-28 14:52:50.830 [info] <0.7.0> Application ejabberd exited with reason: {aborted,{no_exists,passwd,attributes}}
```

**Comment by cromain** [comment]
ejabberd can't access the database, i wonder why. this needs investigation around access right of the process. Start as a service was not really tested. do you know if service process have enough rights for disk i/o ?
also, is there something in ejabberd.log and error.log ?

**Comment by cromain** [comment]
maybe NTFS permissions are not correctly set on these directories. There is an old ticket around starting ejabberd as local service (see #666). I'll try to get hands on a Windows box to check around this. if you have any logs, that would help anyway (you can configure ejabberd to use debug loglevel, and start it as a service)

**Comment by LeoYReyes** [comment]
I ran into a similar problem trying to run start.cmd from a windows service running as the Local System account. I looked at ejabberdctl and saw that it is trying to access the database directory in the home directory. But when a service runs as the Local System account, I think that the home directory points to C:\Windows\System32? So it is actually looking at C:\Windows\system32\config\systemprofile\AppData\Roaming\ejabberd\database\mnesia\? 
When I run the windows service as a different user, the service starts without a problem.

Could this be the issue?
